### PR TITLE
8318959: C2: define MachNode::fill_new_machnode() statically

### PR DESCRIPTION
**Pull request by robcasloz**
This changeset makes the implementation of `MachNode::fill_new_machnode()` static instead of ADLC-generated, for better maintainability. The original motivation to generate this function was to replace its logic with a `ShouldNotCallThis()` call for targets that do not define CISC spill or short branch instruction versions. However, most targets (including x64 and aarch64) feature at least one of these and hence do not make use of the ADLC specialization. For those that do, the savings (approximately 140 bytes on a x64 build of `libjvm.so`) are marginal and (arguably) not worth the additional code complexity.

See motivating discussion in https://mail.openjdk.org/pipermail/hotspot-compiler-dev/2023-October/thread.html#69348.

#### Testing
- tier1-3 (windows-x64, linux-x64, linux-aarch64, macosx-x64, macosx-aarch64; release and debug mode)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8318959](https://bugs.openjdk.org/browse/JDK-8318959): C2: define MachNode::fill_new_machnode() statically (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16444/head:pull/16444` \
`$ git checkout pull/16444`

Update a local copy of the PR: \
`$ git checkout pull/16444` \
`$ git pull https://git.openjdk.org/jdk.git pull/16444/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16444`

View PR using the GUI difftool: \
`$ git pr show -t 16444`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16444.diff">https://git.openjdk.org/jdk/pull/16444.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16444#issuecomment-1788572588)

### CODE REVIEW

File: src/hotspot/share/adlc/output_c.cpp [hunk 1]
```diff
@@ -3249,15 +3212,12 @@ void ArchDesc::defineClasses(FILE *fp) {
       definePeephole(_CPP_PEEPHOLE_file._fp, instr);
 
     // Output code to convert to the cisc version, if applicable
-    used |= instr->define_cisc_version(*this, fp);
+    instr->define_cisc_version(*this, fp);
 
     // Output code to convert to the short branch version, if applicable
-    used |= instr->define_short_branch_methods(*this, fp);
+    instr->define_short_branch_methods(*this, fp);
```
Comment (vnkozlov) [inline src/hotspot/share/adlc/output_c.cpp:3218]: Should you update these 2 methods since their return value is not used anymore anywhere?

Comment (robcasloz) [reply src/hotspot/share/adlc/output_c.cpp:3218]: Good point, done, thanks!

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by robcasloz** [comment]
> Good.

Thanks for reviewing, Vladimir!

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by robcasloz** [comment]
> Looks good.

Thanks, Tobias!

**Pull request closed without merging**